### DISCUSSION TITLE
Prevent editing the object from grid configuration?

### DISCUSSION THREAD

**Discussion by prunepal333** [pimcore/pimcore#13716] [General]
I am unable to find the solution of preventing the user from editing the object from grid configuration.
What I want is only allow the user to edit the object in detail view. 
![Screenshot from 2022-12-01 01-55-53](https://user-images.githubusercontent.com/59438412/204987177-cf282133-766f-4e25-8199-ddcbe7f4940c.png)

**Comment by prunepal333** [comment]
From the above grid, I am able to edit the description and  title field, which needs to be disabled even for admin user.

**Reply by mattamon** [reply]
Hey @prunepal333,

the detail view and the grid view are using the same config. 
I think the easiest way is to disable it by default in the class definition (be aware that in detail view, an admin can always use the "Master (Admin Mode) Layout" to change even non editable fields). It prevents the user and admin to change it in the grid.

![image](https://user-images.githubusercontent.com/38959016/205272782-6a2e5ecd-c95d-4933-985c-c6ced1e3f954.png)


You can then specify a custom layout for the users, so they can edit it in detail view. You can also use this custom layout as default then.

![image](https://user-images.githubusercontent.com/38959016/205273141-538d080d-b418-4f32-8d6b-e903a991c8ec.png)

You can e.g. set the title to editable.

![image](https://user-images.githubusercontent.com/38959016/205273340-a7f6b664-12c9-4f8f-965f-77f88e43681b.png)

**Comment by SamyMP** [comment]
You could do this with an event listener that checks that the opened object is a folder and then immediatly deactivate the input if the user clicks on it :

```js
// startup.js

// After opening a dataobject
document.addEventListener(pimcore.events.postOpenObject, (e) => {
  // FOLDER
  if (e.detail.object.data.general.o_type === "folder") {
    // Getting the panel containing the grid
    const panel = Ext.getCmp(object.search.layout?.id);

    if (!panel) {
      return;
    }

    // Will execute when the grid appears in the dom
    const mutations = new MutationObserver((mutation) => {
      if (panel && panel.el.dom && mutation[0].addedNodes[0]?.id?.startsWith("celleditor")) {
        mutation[0].addedNodes[0].querySelector('input').disabled = true;
      }

      // TODO : handle boolean type, relation type...
    })

    mutations.observe(panel.el.dom, {
      childList: true,
      subtree: true,
    })
  }
});
```

This should give you a good start
